scripts/check_clean_enlistment.sh

```diff
@@ -12,11 +12,9 @@
 #
 # Examples: scripts/check_clean_enlistment.sh
 
-# Treat travis builds as 'clean'
-if [ -z "${TRAVIS_BRANCH}" ]; then
-    if [[ -n $(git status --porcelain) ]]; then
-        exit 1
-    fi
+# Check for clean enlistment ignoring the update to buildnumber.dat which is expected during a release build
+if (( $(git status --porcelain | grep -v "buildnumber.dat" | wc -l) == 0 )); then
```

Comment:
The `wc` in the pipeline isn't necessary, and you can just check the exit code in the branch to make this much simpler.
But, please see my suggestion about moving this to `checkout_branch.sh`.  It makes it much simpler.

Also, you'll want to install shellcheck to lint the script files that you work on, as it stands I believe the scripts would be generating errors.

Reply:
No errors, but shellcheck suggested using grep -c in place of wc -l.  I made that change.